torch/testing/_internal/opinfo/core.py

```diff
@@ -1199,6 +1205,59 @@ def get_inplace_operator(self):
         Returns None if the operator has no inplace operator variant"""
         return self.inplace_operator_variant
 
+    # Returns a callable from TestCase -> subtest context manager xfailing / skipping only
+    # for expected errors.
+    def _maybe_skip_or_xfail(self, rules, device, sample, idx):
+        def _subtest_fn(test_case, sample=sample, idx=idx):
+            return test_case.subTest(sample=sample, idx=idx)
+
+        if rules is None or len(rules) == 0:
+            return _subtest_fn
+
+        # NB: match first rule only (order matters!)
+        for rule in rules:
+            if rule.sample_match_fn(device, sample):
+                log.debug(
+                    "matched %s rule '%s': %s %s %s",
+                    rule.type,
+                    rule.name,
+                    self.full_name,
+                    device,
+                    sample,
+                )
+
+                # combined subtest + rule-specific context manager
+                return lambda test_case: SubtestRuleCtx(
```

Comment:
for my understanding, is SubtestRuleCtx either a fail or a skip? Can we document that?